app/src/main/java/com/example/eric/socials/AddEvent.java

```diff
@@ -0,0 +1,133 @@
+                    }
+                }).addOnSuccessListener(new OnSuccessListener<UploadTask.TaskSnapshot>() {
+                    @Override
+                    public void onSuccess(UploadTask.TaskSnapshot taskSnapshot) {
+                        ArrayList<String> users = new ArrayList<String>();
+                        users.add(mUser.getUid());
+                        Social s = new Social(eventName.getText().toString(), key + ".png", mUser.getEmail() , description.getText().toString(),
+                                1, date.getText().toString(), users);
+                        myRef.child(key).setValue(s);
+                        Toast.makeText(getApplicationContext(), "Post Successful!", Toast.LENGTH_SHORT).show();
+                        Intent intent = new Intent(getApplicationContext(), FeedActivity.class);
+                        startActivity(intent);
+                    }
+                });
+                Utils.progressBar(this, getString(R.string.create_social_progressbar));
+                break;
+            default:
+        }
+    }
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_add_event);
+
+        eventName = (EditText) findViewById(R.id.eventName);
+        date = (EditText) findViewById(R.id.eventDate);
+        description = (EditText) findViewById(R.id.description);
+        createSocial = (Button) findViewById(R.id.createEventButton);
+        eventImage = (ImageView) findViewById(R.id.eventImage);
+
+        eventImage.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                AlertDialog alertDialog = new AlertDialog.Builder(AddEvent.this).create();
+                alertDialog.setTitle("Set a Photo");
+                alertDialog.setButton(AlertDialog.BUTTON_NEGATIVE, "Upload from Gallery",
+                        new DialogInterface.OnClickListener() {
+                            public void onClick(DialogInterface dialog, int which) {
+                                //launch gallery
+                                startActivityForResult(new Intent(Intent.ACTION_PICK, android.provider.MediaStore.Images.Media.INTERNAL_CONTENT_URI), 3);
+                                dialog.dismiss();
+                            }
+                        });
+                alertDialog.show();
+            }
+        });
+
+        mAuth = FirebaseAuth.getInstance();
+        mUser = mAuth.getCurrentUser();
+
+        createSocial.setOnClickListener(this);
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        //Detects request codes
+        if ((requestCode == 3 || requestCode == 1) && resultCode == Activity.RESULT_OK) {
```

Comment:
What do 3 and 1 represent? Define them as public static final ints at the top so we know what request each corresponds to. In the 61B style guide, these are referred to as "magic numbers"